### DISCUSSION TITLE
Suggestions for how to add tooltips

### DISCUSSION THREAD

**Discussion by logularjason** [observablehq/plot#966] [Q&A] [answered]
Thanks for building this promising framework.

Seeking advice on whether it would be possible for me to create a chart tooltips as shown in the screenshot below.  We have a use-case where these tooltips need to appear on all plots as the user pans left and right.  If it helps, I am very experienced with d3 - I just need a nudge in the right direction if there is a way I can add this custom functionality assuming it's not native yet. 

![image](https://user-images.githubusercontent.com/53684229/177001267-011af71c-7c59-4d8e-bb80-4d5d81ec5b6a.png)

**Comment by enjalot** [comment, marked as answer]
One approach for adding tooltips is implemented here: https://observablehq.com/@mkfreeman/plot-tooltip
At the moment it doesn't support custom html in the tooltip, but if you're experienced with d3 you can probably see how to extend it with that functionality.

**Reply by Fil** [reply]
And here's an alternative prototype, that uses tippy.js to create elaborate tooltips:
https://observablehq.com/@fil/plot-tippy-js-2 (based on https://observablehq.com/@fil/hello-tippy ).

All this is in a "prototype" stage, we're still tinkering with the possibilities. We hope to be able to have a supported approach in the future for tooltips/popups and other interactions.

**Reply by logularjason** [reply]
Thankyou both for the tips - makes sense.

Hoping you can give me another nudge - the thrust of the question was to show the tooltips for ALL plots on the chart at once as shown.  Kind of like the user is moving a cursor left and right along the chart.  Is this beyond what is possible (I'm happy to code it if there is a point that I can hook in).
